tests/tobjdefs.nim

```diff
@@ -126,3 +126,34 @@ suite "Object Api":
     check objDef(MyType())
     check objDef(MyType)
 
+  test "Get ObjectDef (generic)":
+    macro objDef(t: typed): bool =
+      discard objectDef(t)
+      newLit true
+    type MyType[T] = object
+
+    var myVar = MyType[int]()
+    let myLet = MyType[int]()
+
+    check objDef(myVar)
+    check objDef(myLet)
+    check objDef(MyType[int]())
+    check objDef(MyType)
+    # check objDef(MyType[int]) # unfortunately, this doesn't work
```

Comment:
Yea this is a compiler restriction, which https://github.com/nim-lang/Nim/pull/22432 aimed to fix, but alas I'm too daft.